### ISSUE TITLE
certificate alerting calculates back in hours instead of days

### ISSUE THREAD

**Issue by ThomasLD** [codebude/cpi-dashboard#4]
Hi,

Found a small issue when we noticed certificates that were expiring in less than 2 weeks did not trigger an alert mail. I'm assuming of course you would calculate the amount of days back as said in the front-end but in https://github.com/codebude/cpi-dashboard/blob/master/IFlow/Source/src/main/resources/script/alertCalculateAlerts.groovy on line 67 it calculates with hours so instead of checking for 14 days it checks for 14 hours until expiry of the certificate.

KR,
Thomas

**Comment by codebude** [comment]
Thanks for the PR. I merged the changes, thus they will be part of the next release. Meanwhile feel free to build the dashboard from the codebase provided here on Github.

**Comment by janengelmohr** [comment]
Already done that (https://github.com/engelmohr/cpi-dashboard/commit/e9b42c93d3f59d3b1484f5e40772f7d6a5e922d6). I can send you another PR with the built zip file if you like. 

Also since lots of things for the setup have changed (e.g. no more unofficial APIs needed and thus no more basic auth user) I could also update the README/HOWTO if you want. Just let me know what you think.


Best regards,
Jan

**Comment by codebude** [comment]
Hi @engelmohr ,

you're right. The readme (i guess you mean this blog post: https://blogs.sap.com/2019/08/19/advanced-monitoring-and-health-check-with-realcores-cpi-dashboard/ ) doesn't reflect the current codebase in this repository. The blog still reflects the latest release (1.0.4): https://github.com/codebude/cpi-dashboard/releases

I guess updating the readme makes only sense, when also publishing the current version (1.0.5) of the dashboard as new release. But before doing so, we should see that the current codebase is strong enough for a stable build. Currently I get some 500er errors for some of the endpoints.

![image](https://user-images.githubusercontent.com/2236234/100129281-a4793d80-2e81-11eb-8ca6-c06e0525f02d.png)

Let me/us first fix them, before packaging and releasing 1.0.5 and in consequence of this updating the readme. Do those errors occur for you, too?

**Comment by codebude** [comment]
One more side note, @engelmohr . You wrote:

> [...] e.g. no more unofficial APIs needed and thus no more basic auth user [...]

I fear that's not correct. The basic auth user is still needed for the following features:

- List endpoints: https://github.com/codebude/cpi-dashboard/blob/master/IFlow/Source/src/main/resources/script/loadEndpointList.groovy#L42
- Get development packages:
-- For Diffutils: https://github.com/codebude/cpi-dashboard/blob/master/IFlow/Source/src/main/resources/script/diffCompareIFlows.groovy#L79
-- For version comparison: https://github.com/codebude/cpi-dashboard/blob/master/IFlow/Source/src/main/resources/script/remoteTenantArtifactsComparison.groovy#L132

For all of those there are official APIs now (https://api.sap.com/api/IntegrationContent/resource). So it should be possible to avoid the use of the basic auth credentials in the future. But right now, the basic auth credentials are still needed.

**Comment by janengelmohr** [comment]
Hi @codebude,

yes, I mean your SAP blog post :)

So I just checked on our CPI and I do not see any 500 errors whatsoever for none of the endpoints. I will double-check and do a fresh re-pack of the current source as of commit https://github.com/codebude/cpi-dashboard/commit/269c701af1ad3acecd3b09a66c80fe8a3d52caed to make sure I did not miss anything.

**Comment by codebude** [comment]
@engelmohr  The 500 http errors were my fault. Some old password in the basic auth credential... At least this shows, that the basic auth credentials are still needed. ;-) So before releasing the current version, I/we should eliminate the usage of those unofficial APIs and thus ensure that no basic auth credentials are needed any longer.

After that, I'm fine with releasing the a new version.

**Comment by janengelmohr** [comment]
You're right. I missed those instances of basic auth. I somehow thought it's not needed anymore. Anyways I will refactor this within this week given you don't want to do it yourself ;-)


Edit: I just browsed the code and seems like basic auth is way more often used than I anticipated. Gonna start working on this now but since this is a fun spare time project I would like to revoke the given deadline from above :D
